cpp/daal/src/threading/threading.cpp

```diff
@@ -111,6 +111,33 @@ DAAL_EXPORT void _daal_threader_for(int n, int threads_request, const void * a,
 #endif
 }
 
+DAAL_EXPORT void _daal_static_threader_for(size_t n, const void * a, daal::functype_static func)
+{
+#if defined(__DO_TBB_LAYER__)
+    const size_t nthreads           = _daal_threader_get_max_threads();
+    const size_t nblocks_per_thread = n / nthreads + !!(n % nthreads);
+
+    tbb::parallel_for(
+        tbb::blocked_range<size_t>(0, nthreads, 1),
```

Comment:
In this case, the block size will be very large? Are we losing gains in caches and load balancing threads?

Reply:
We still process the work by blocks, so caching is the same.

Work balancing is static now, not dynamic as was before. It can bring perf improvements, as well as regressions. Need to look case by case.

Reply:
Has there been any improvement anywhere?